### PR TITLE
RDK-38312 : Wake up source APIS in system services

### PR DESCRIPTION
**Pull request by akhilsarada**
Reason for change: wakeup source Thunder implementation
Test Procedure: Refer JIRA
Risks: Low
Priority : p2
Signed-off-by: Akhil Baby Sarada <Akhil_Sarada@comcast.com>

### CODE REVIEW

File: SystemServices/SystemServices.cpp [hunk 1]
```diff
@@ -3761,55 +3759,192 @@ namespace WPEFramework {
...
+                    powerState = (1<<IARM_BUS_PWRMGR_POWERSTATE_STANDBY_LIGHT_SLEEP) | \
+                                             (1 << IARM_BUS_PWRMGR_POWERSTATE_STANDBY_DEEP_SLEEP);
+            }
+            LOGWARN("%s: %d Power State stored:%x \r\n",__FUNCTION__,__LINE__,powerState);
+            if (parameters.HasLabel("WAKEUPSRC_VOICE")) 
+            {
+                srcType |= 1<<WAKEUPSRC_VOICE;
+                if(parameters["WAKEUPSRC_VOICE"].Boolean())
+                    config |= 1<<WAKEUPSRC_VOICE;
+                LOGWARN("%s: %d WAKEUPSRC_VOICE srcType:%x  config :%x \n",__FUNCTION__,__LINE__,srcType ,config);
```
Comment (apatel859) [inline SystemServices/SystemServices.cpp:3787]: could you move this at end as it is repeated in every if else

File: SystemServices/SystemServices.cpp [hunk 2]
```diff
@@ -3761,55 +3759,192 @@ namespace WPEFramework {
...
+            IARM_Result_t res = IARM_Bus_Call(IARM_BUS_PWRMGR_NAME,
+                                  IARM_BUS_PWRMGR_API_GetWakeupSrcConfig, (void *)&param,
+                                  sizeof(param));
+            if (IARM_RESULT_SUCCESS == res) {
+               LOGWARN(" %s: %d res:%d srcType :%x  config :%x \n",__FUNCTION__,__LINE__,res,param.srcType,param.config);
+               status = true;
+               if( param.srcType & (1<<WAKEUPSRC_VOICE))
+               {
+                    response["WAKEUPSRC_VOICE"] = (param.config & (1<<WAKEUPSRC_VOICE))?true:false;
+                    LOGWARN(" %s: %d  WAKEUPSRC_VOICE value:%d  \n",__FUNCTION__,__LINE__,param.config & (1<<WAKEUPSRC_VOICE));
```
Comment (apatel859) [inline SystemServices/SystemServices.cpp:3883]: no need to print here as combined value already printed in line 3878

File: SystemServices/SystemServices.cpp (hunk 1 above)
Comment (akhilsarada) [reply SystemServices/SystemServices.cpp]: removed log lines to common place

File: SystemServices/SystemServices.cpp (hunk 2 above)
Comment (akhilsarada) [reply SystemServices/SystemServices.cpp]: removed

**Review by apatel859 — APPROVED** [review]

**Comment by anand-ky** [comment]
Closing older PRs. Pls bring to latest sprint or topic branch.

**Pull request closed without merging**